### DISCUSSION TITLE
How to get all conversations data?

### DISCUSSION THREAD

**Discussion by ghafarizar** [tulir/whatsmeow#422] [WhatsApp protocol Q&A]
As title explained, how can I get all conversations from a number?
https://pkg.go.dev/go.mau.fi/whatsmeow#Client.ParseWebMessage
I know this link explained them, but still i cannot figure out how.

```
chatJID, err := types.ParseJID(conv.GetId())
for _, historyMsg := range conv.GetMessages() {
	evt, err := cli.ParseWebMessage(chatJID, historyMsg.GetMessage())
	yourNormalEventHandler(evt)
}
```

like, where is 'conv' come from? and whats historyMsg is?

**Comment by hoehermann** [comment]
The conversations are pushed by the server during logon. I am using it [here](https://github.com/hoehermann/purple-gowhatsapp/blob/7f19e2b5bcdbb2f73ed315b9af5387f21187c456/src/go/handler.go#L105). As far as I know, there is no way to request more data for a specific conversation. Even the official WhatsApp Web interface has the same limitation: It shows a message "Use WhatsApp on your phone to see earlier messages".
